pkg/util/scrubber/scrubber.go

```diff
@@ -213,11 +214,44 @@ func (c *Scrubber) scrub(data []byte, replacers []Replacer) []byte {
 		}
 		if len(repl.Hints) == 0 || containsHint {
 			if repl.ReplFunc != nil {
-				data = repl.Regex.ReplaceAllFunc(data, repl.ReplFunc)
+				data = repl.Regex.ReplaceAllFunc(data, func(match []byte) []byte {
+					if containsValidENC(match) {
+						return match // Don't replace if contains valid ENC[]
+					}
+					return repl.ReplFunc(match)
+				})
 			} else {
-				data = repl.Regex.ReplaceAll(data, repl.Repl)
+				data = repl.Regex.ReplaceAllFunc(data, func(match []byte) []byte {
+					if containsValidENC(match) {
+						return match // Don't replace if contains valid ENC[]
+					}
+					return repl.Regex.ReplaceAll(match, repl.Repl)
+				})
 			}
 		}
 	}
 	return data
 }
+
+// IsEnc returns true is the string match the 'ENC[...]' format
+// Borrowed from comp/core/secrets/utils
+func IsEnc(str string) bool {
+	// trimming space and tabs
+	str = strings.Trim(str, " \t")
+	if strings.HasPrefix(str, "ENC[") && strings.HasSuffix(str, "]") {
+		return true
+	}
+	return false
+}
+
+// containsValidENC checks if the match contains a valid ENC[]
+func containsValidENC(data []byte) bool {
```

Comment:
I'm not sure I understand the difference between `IsEnc` and `ContainsValidEnc`. Are we no scrubbing the data if it contains at least one ENC ? What happens if something container `api_key=1234 app_key=ENC[something]` ?

Could you update the docstring to explain it ?